### PR TITLE
fix: [CDS-34383]: Showing appropriate error for addService via YAML and coverage improvement

### PR DESCRIPTION
**Pull request by Sagar-Rajput7**
##### Summary:

> Corrected the error message
> Also added tests and coverage improvement

##### Jira Links:

https://harness.atlassian.net/browse/CDS-34383

##### Screenshots:

Earlier : -

Screenshot 2022-03-08 at 10 31 12 AM

Now :- 

Screenshot 2022-03-08 at 10 30 12 AM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by dhruvUpadhyay07 — APPROVED** [review]

File: src/modules/75-cd/components/Environments/EnvironmentList/EnvironmentsModal.tsx [hunk 1]
```diff
@@ -152,17 +170,11 @@ export const NewEditEnvironmentModalYaml: React.FC<NewEditEnvironmentModalProps>
   const handleModeSwitch = React.useCallback(
     (view: SelectedView) => {
       if (view === SelectedView.VISUAL) {
-        const yaml = yamlHandler?.getLatestYaml() || /* istanbul ignore next */ ''
+        const yaml = defaultTo(yamlHandler?.getLatestYaml(), /* istanbul ignore next */ '')
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/Environments/EnvironmentList/EnvironmentsModal.tsx:173]: We don't need this /* istanbul ignore next */ if we are using defaultTo

File: src/modules/75-cd/components/Environments/EnvironmentList/EnvironmentsModal.tsx [hunk 2]
```diff
@@ -255,7 +267,7 @@ export const NewEditEnvironmentModalYaml: React.FC<NewEditEnvironmentModalProps>
                         type="submit"
                         text={getString('save')}
                         onClick={() => {
-                          const latestYaml = yamlHandler?.getLatestYaml() || /* istanbul ignore next */ ''
+                          const latestYaml = defaultTo(yamlHandler?.getLatestYaml(), /* istanbul ignore next */ '')
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/Environments/EnvironmentList/EnvironmentsModal.tsx:270]: We don't need this /* istanbul ignore next */ if we are using defaultTo

File: src/modules/75-cd/components/Environments/EnvironmentList/EnvironmentsModal.tsx (hunk 1 above)
Comment (Sagar-Rajput7) [reply src/modules/75-cd/components/Environments/EnvironmentList/EnvironmentsModal.tsx:173]: ok I will remove it

Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/Environments/EnvironmentList/EnvironmentsModal.tsx:173]: You can let this be for now. Next time, we can remove it

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/75-cd/components/Environments/EnvironmentList/EnvironmentsModal.tsx (hunk 2 above)
Comment (Sagar-Rajput7) [reply src/modules/75-cd/components/Environments/EnvironmentList/EnvironmentsModal.tsx:270]: Resolved

**Pull request merged**